File: gateway/src/models/auth/auth_contract.go
package auth

import (
	"time"

	commsec "gateway/src/models/commsec"

	"github.com/google/uuid"
)

// TokenIssueRequest 表示签发任意类型令牌的统一参数。
type TokenIssueRequest struct {
	Principal Principal

	TokenType TokenType
	SessionID uuid.UUID
	FamilyID  uuid.UUID

	Audience string
	Role     string
	Scopes   []string

	AuthMethod AuthMethod

	ClientID      string
	GatewayID     string
	SourceService string
	TargetService string

	ParentTokenID uuid.UUID
	TTLSec        int64
}

// TokenRefreshRequest 表示 refresh token 刷新参数。
type TokenRefreshRequest struct {
	RefreshToken string

	ClientID  string
	GatewayID string
	SourceIP  string
	UserAgent string

	RequestID string
	TraceID   string
}

// TokenVerifyRequest 表示令牌校验参数。
type TokenVerifyRequest struct {
	RawToken string

	ExpectedTypes    []TokenType
	ExpectedAudience string
	RequireScopes    []string

	SourceService string
	TargetService string

	AllowExpiredSkewSec int64
}

// TokenRevokeRequest 表示令牌撤销参数。
type TokenRevokeRequest struct {
	TokenID   uuid.UUID
	FamilyID  uuid.UUID
	SessionID uuid.UUID

	Reason    string
	RevokedBy string

	RequestID string
	TraceID   string
}

// SessionIssueRequest 表示创建会话参数。
type SessionIssueRequest struct {
	Principal Principal
	Role      string
	Scopes    []string

	AuthMethod AuthMethod

	ClientID  string
	GatewayID string
	SourceIP  string
	UserAgent string

	ExpiresAt time.Time
}

// SessionValidateRequest 表示会话校验参数。
type SessionValidateRequest struct {
	SessionID     uuid.UUID
	PrincipalID   string
	RequireActive bool
	MinVersion    int64
}

// SessionRevokeRequest 表示会话撤销参数。
type SessionRevokeRequest struct {
	SessionID   uuid.UUID
	PrincipalID string

	Reason    string
	RevokedBy string

	RequestID string
	TraceID   string
}

// DownstreamGrantRequest 表示网关申请下游服务授权参数。
type DownstreamGrantRequest struct {
	Identity IdentityContext

	TargetService string
	BindingType   commsec.ChannelBindingType

	RequireEncryption bool
	TTLSec            int64
}

File: api_service/src/models/auth/auth_contract.py
from msgspec import Struct

from src.models.auth.auth import (
    AuthMethod,
    IdentityContext,
    Principal,
    TokenType,
)
from src.models.commsec.commsec import ChannelBindingType


class TokenIssueRequest(Struct, kw_only=True):
    """表示签发任意类型令牌的请求。"""

    principal: Principal

    token_type: TokenType
    session_id: str
    family_id: str

    audience: str
    role: str
    scopes: list[str]

    auth_method: AuthMethod

    client_id: str
    gateway_id: str
    source_service: str
    target_service: str

    parent_token_id: str
    ttl_sec: int


class TokenRefreshRequest(Struct, kw_only=True):
    """表示刷新令牌的请求。"""

    refresh_token: str

    client_id: str
    gateway_id: str
    source_ip: str
    user_agent: str

    request_id: str
    trace_id: str


class TokenVerifyRequest(Struct, kw_only=True):
    """表示令牌校验参数。"""

    raw_token: str

    expected_types: list[TokenType]
    expected_audience: str
    require_scopes: list[str]

    source_service: str
    target_service: str

    allow_expired_skew_sec: int


class TokenRevokeRequest(Struct, kw_only=True):
    """表示令牌撤销参数。"""

    token_id: str
    family_id: str
    session_id: str

    reason: str
    revoked_by: str

    request_id: str
    trace_id: str


class SessionIssueRequest(Struct, kw_only=True):
    """表示创建会话参数。"""

    principal: Principal
    role: str
    scopes: list[str]

    auth_method: AuthMethod

    client_id: str
    gateway_id: str
    source_ip: str
    user_agent: str

    expires_at: float


class SessionValidateRequest(Struct, kw_only=True):
    """表示会话校验参数。"""

    session_id: str
    principal_id: str
    require_active: bool
    min_version: int


class SessionRevokeRequest(Struct, kw_only=True):
    """表示会话撤销参数。"""

    session_id: str
    principal_id: str

    reason: str
    revoked_by: str

    request_id: str
    trace_id: str


class DownstreamGrantRequest(Struct, kw_only=True):
    """表示网关申请下游服务授权参数。"""

    identity: IdentityContext

    target_service: str
    binding_type: ChannelBindingType

    require_encryption: bool
    ttl_sec: int
